src/main/scala/org/broadinstitute/hail/stats/SparseGtBuilder.scala

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hail.stats
+
+import breeze.linalg.SparseVector
+import org.broadinstitute.hail.variant.Genotype
+
+import scala.collection.mutable
+
+// constructs SparseVector of genotype calls with missing values mean-imputed
+// if all genotypes are missing then all elements are NaN
+class SparseGtBuilder extends Serializable {
```

Comment:
SparseGtBuilder is more generic and only depends on genotypes, whereas LinRegBuilder is optimized to precompute sufficient stats like xx and xy needed only in case of straight up linear regression.  I think I should return to this when unifying what the regression models output (right now LMM gives additional counts like nHet with respect to included samples only, but if may make more sense to filterSamples on the vds at the beginning, will discuss more in person).